### PR TITLE
[New Rule] AWS STS GetSessionToken Abuse

### LINKED ISSUE #1152: [New Rule] AWS STS GetSessionToken Abuse
## Description
Identifies the suspicious use of GetSessionToken. Tokens could be created and used by attackers to move laterally and escalate privileges.

### Target indexes
`filebeat-*, logs-aws*`

### Platforms
AWS Cloudtrail

### Query

```
event.dataset:aws.cloudtrail and 
event.provider:sts.amazonaws.com and 
event.action:GetSessionToken and aws.cloudtrail.user_identity.type:IAMUser and
event.outcome:success
```

```
event.dataset:aws.cloudtrail and event.provider:sts.amazonaws.com and event.action:GetSessionToken and aws.cloudtrail.user_identity.type:IAMUser and event.outcome:success
```

> What is the user type for just a plain IAM user?

### Related issues or PRs
- [#955 ](https://github.com/elastic/detection-rules/issues/955)

### **False Positives**
- 
- 

### MITRE

| Tactic | Technique ID | Technique Name | Sub-Technique Name |
| ------ | ------------ | -------------- | ------------------ |
|Privilege Escalation|T1548|Abuse Elevation Control Mechanism|                    |
|Lateral Movement|T1550|Use Alternate Authentication Material|Application Access Token|

`lateral_movement_sts_getsessiontoken_abuse.toml`

### References
- 

### Tags

`Elastic,Cloud,AWS,Continuous Monitoring, SecOps,Identity and Access`

### PR DESCRIPTION
**Pull request by austinsonger**
## Issues

Resolves #1152 
Relates #955 

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -8,6 +8,7 @@
 /*.yml
 /config
 /junit.xml
+/elastic-detection-cli
```
Comment (brokensound77) [inline .gitignore:11]:
```suggestion
```

File: rules/aws/privilege_escalation_sts_getsessiontoken_abuse.toml [hunk 2]
```diff
@@ -0,0 +1,62 @@
+[metadata]
+creation_date = "2021/05/17"
+maturity = "development"
```
Comment (brokensound77) [inline rules/aws/privilege_escalation_sts_getsessiontoken_abuse.toml:3]:
```suggestion
maturity = "production"
```

File: rules/aws/privilege_escalation_sts_getsessiontoken_abuse.toml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+updated_date = "2021/05/17"
+
+[rule]
+author = ["Elastic", "Austin Songer"]
+description = """
+Identifies the suspicious use of GetSessionToken. Tokens could be created and used by attackers to move laterally and 
+escalate privileges.
+"""
+index = ["filebeat-*", "logs-aws*"]
```
Comment (brokensound77) [inline rules/aws/privilege_escalation_sts_getsessiontoken_abuse.toml:12]: can you add `false_positives` and `references` please

Comment (austinsonger) [reply rules/aws/privilege_escalation_sts_getsessiontoken_abuse.toml:12]: @brokensound77 yeah sorry. I was meaning to come back to this and finish it.

File: rules/integrations/aws/privilege_escalation_sts_getsessiontoken_abuse.toml [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+false_positives = [
+    """
+    GetSessionToken may be done by a system or network administrator. Verify whether the user identity, user 
+    agent, and/or hostname should be making changes in your environment. GetSessionToken from unfamiliar users or 
+    hosts should be investigated. If known behavior is causing false positives, it can be exempted from the rule.
+    """,
+]
+index = ["filebeat-*", "logs-aws*"]
+language = "kuery"
+name = "AWS STS GetSessionToken Abuse"
```
Comment (w0rk3r) [inline rules/integrations/aws/privilege_escalation_sts_getsessiontoken_abuse.toml:21]:
```suggestion
language = "kuery"
license = "Elastic License v2"
name = "AWS STS GetSessionToken Abuse"
```

Comment (w0rk3r) [reply rules/integrations/aws/privilege_escalation_sts_getsessiontoken_abuse.toml:21]: Missing License

**Review by w0rk3r — CHANGES_REQUESTED** [review]
License needs to be added, other than that, LGTM

**Review by brokensound77 — APPROVED** [review]
LGTM, thanks

**Review by w0rk3r — APPROVED** [review]

**Pull request merged**